com.unity.multiplayer.mlapi/Prototyping/NetworkTransform.cs

```diff
@@ -12,282 +14,180 @@ namespace MLAPI.Prototyping
     public class NetworkTransform : NetworkBehaviour
     {
         /// <summary>
-        /// Server authority only allows the server to update this transform
-        /// Client authority only allows the client owner to update this transform
+        /// Server authority allows only the server to update this transform
+        /// Client authority allows only the owner client to update this transform
         /// Shared authority allows everyone to update this transform
         /// </summary>
-        public enum Authority
+        public enum NetworkAuthority
         {
-            Server = 0, // default
+            Server = 0,
             Client,
             Shared
         }
 
-        /// <summary>
-        /// TODO this will need refactoring
-        /// Specifies who can update this transform
-        /// </summary>
-        [SerializeField, Tooltip("Defines who can update this transform.")]
-        public Authority TransformAuthority = Authority.Server; // todo Luke mentioned an incoming system to manage this at the NetworkBehaviour level, lets sync on this
-
-        /// <summary>
-        /// The base amount of sends per seconds to use when range is disabled
-        /// </summary>
-        [SerializeField, Range(0, 120), Tooltip("The base amount of sends per seconds to use when range is disabled")]
-        public float FixedSendsPerSecond = 30f;
-
-        /// <summary>
-        /// TODO MTT-766 once we have per var interpolation
-        /// Enable interpolation
-        /// </summary>
-        [SerializeField, Tooltip("This requires AssumeSyncedSends to be true")]
-        public bool InterpolatePosition = true;
-
-        /// <summary>
-        /// TODO MTT-766 once we have per var interpolation
-        /// The distance before snaping to the position
-        /// </summary>
-        [SerializeField, Tooltip("The transform will snap if the distance is greater than this distance")]
-        public float SnapDistance = 10f;
-
-        /// <summary>
-        /// TODO MTT-766 once we have per var interpolation
-        /// Should the server interpolate
-        /// </summary>
-        [SerializeField]
-        public bool InterpolateServer = true;
+        private class NetworkState : INetworkSerializable
```

Comment:
should be struct

Reply:
`class` is intentional here as we're modifying its props in `UpdateNetworkState()` instead of swapping `m_NetworkState.Value` with a brand-new one

Reply:
I still think this should be a struct, also just to get rid of the allocations when deserializing this into a new instance. But we can just go ahead with `class` that's fine.

Reply:
Since this won't be used by users, only us internally, I'd be with Luke on that one and avoid the allocation vs having something more friendly.
From what I understand, we're creating a new instance each time we deserialize a type.

Reply:
I don't think we create new instances everytime we deserialize a type here. why do you think that's the case @LukeStampfli ?

Reply:
Because NetworkVariable ReadDelta is a bit messy.
- It does not read a delta but an absolute value
- It uses `ReadObjectPacked` which internally creates a new instance of `NetworkState`
- It also boxes the value an additional time in `ReadObjectPacked` when returning it, if it is a struct.
https://github.com/Unity-Technologies/com.unity.multiplayer.mlapi/blob/ee59d5fb560bbd6273260c1df144172914b38448/com.unity.multiplayer.mlapi/Runtime/NetworkVariable/NetworkVariable.cs#L187

So it looks like we allocate anyways no matter whether we use a struct or class here 😞